server/app.js

```diff
@@ -63,4 +189,219 @@ io.on('err', function(err){
 
 http.listen(port,function(){
     log.info('Server listening through port %s', port);
-});
\ No newline at end of file
+});
+
+http.on('err', function(err){
+   // handle the error safely
+    console.log(err)
+});
+
+process.on('uncaughtException', function(err) {
+  console.log('Caught exception: ' + err);
+});
+
+
+var state = ["NAVIGATING_SIDEWALK", "LOOKING_FOR_PATH" ,"FOUND_PATH_TO_HOUSE", "NAVIGATING_PATH" ,"REACHED_DOOR" , "DELIVER_PACKAGE" ]
```

Comment:
Const?

Reply:
Yeah, this will be changed to be a dict as well so constant enums can be used